src/dbnode/persist/fs/seek_manager.go

```diff
@@ -208,7 +213,16 @@ func (m *seekerManager) Open(
 }
 
 func (m *seekerManager) CacheShardIndices(shards []uint32) error {
-	multiErr := xerrors.NewMultiError()
+	var (
+		multiErr    = xerrors.NewMultiError()
+		resultsLock sync.Mutex
+		wg          sync.WaitGroup
+	)
+
+	// NB(r): Since this is mainly IO bound work, perfectly
+	// find to do this in parallel.
+	workers := xsync.NewWorkerPool(concurrentCacheShardIndices)
```

Comment:
Do we want to spin up a worker pool every time `CacheShardIndices` is called?

Reply:
Can reuse this.

Reply:
Fixed this up 👍